## Statistical Analysis of Study Hours vs. Scores

In [11]:
import pandas as pd

In [ ]:
#Load Dataset
df = pd.read_csv('hours_and_score.csv')
df.tail(5)

,Hours,Scores
20,2.7,30
21,4.8,54
22,3.8,35
23,6.9,76
24,7.8,86


# 1. Calculate Means

In [16]:
n = len(df)
x = df.Hours
y = df.Scores
x_sum = x.sum()
y_sum = y.sum()
x_mean = x_sum/n
y_mean = y_sum/n
print("Mean X:", x_mean)
print("Mean Y:", y_mean)

Mean X: 5.012
Mean Y: 51.48


The average study time is 5.01 hours, resulting in an average score of 51.48.

# 2. Calculate Variance

In [24]:
x_variance = ((x - x_mean)**2).sum()/(n-1)
y_variance = ((y - y_mean)**2).sum()/(n-1)
print("Variance X:", x_variance)
print("Variance Y:", y_variance)

Variance X: 6.3761
Variance Y: 639.4266666666666


A variance of 6.38 in hours and 639.43 in scores shows how much the data spread out from their respective means.

# 3. Calculate Covariance

In [25]:
covariance = (((x - x_mean)*(y - y_mean)).sum())/(n-1)
print("Covariance:", covariance)

Covariance: 62.331500000000005


The positive covariance of 62.33 indicates that as study hours increase, scores generally tend to increase as well.

# 4. Calculate Standard Deviation

In [27]:
x_std = x_variance ** 0.5
y_std = y_variance ** 0.5
print("Standard Deviation X:", x_std)
print("Standard Deviation Y:", y_std)

Standard Deviation X: 2.5250940576540906
Standard Deviation Y: 25.28688724747802


# 5. Calculate Correlation

In [28]:
correlation = covariance/(x_std*y_std)
print("Correlation:", correlation)

Correlation: 0.9761906560220888


The correlation coefficient of roughly 0.976 signifies a very strong positive linear relationship

### **Conclusion**

This analysis shows that there is a positive correlation between study hours and scores. This means that as the number of study hours increases, the scores tend to increase as well. This suggests that studying more hours can lead to better academic performance.